plugin/src/main/java/org/opensearch/ml/action/models/UpdateModelTransportAction.java

```diff
@@ -180,29 +210,50 @@ private void updateRemoteOrTextEmbeddingModel(
         MLUpdateModelInput updateModelInput,
         MLModel mlModel,
         User user,
-        ActionListener<UpdateResponse> actionListener,
-        ThreadContext.StoredContext context
+        ActionListener<UpdateResponse> wrappedListener,
+        boolean isModelDeployed
     ) {
         String newModelGroupId = (Strings.hasLength(updateModelInput.getModelGroupId())
             && !Objects.equals(updateModelInput.getModelGroupId(), mlModel.getModelGroupId())) ? updateModelInput.getModelGroupId() : null;
         String relinkConnectorId = Strings.hasLength(updateModelInput.getConnectorId()) ? updateModelInput.getConnectorId() : null;
 
         if (mlModel.getAlgorithm() == TEXT_EMBEDDING) {
-            if (relinkConnectorId == null) {
-                updateModelWithRegisteringToAnotherModelGroup(modelId, newModelGroupId, user, updateModelInput, actionListener, context);
+            if (relinkConnectorId == null && updateModelInput.getConnectorUpdateContent() == null) {
+                updateModelWithRegisteringToAnotherModelGroup(
+                    modelId,
+                    newModelGroupId,
+                    user,
+                    updateModelInput,
+                    wrappedListener,
+                    isModelDeployed
+                );
             } else {
-                actionListener
+                wrappedListener
                     .onFailure(
                         new OpenSearchStatusException(
-                            "Trying to update the connector or connector_id field on a local model",
+                            "Trying to update the connector or connector_id field on a local model.",
                             RestStatus.BAD_REQUEST
                         )
                     );
             }
         } else {
             // mlModel.getAlgorithm() == REMOTE
             if (relinkConnectorId == null) {
-                updateModelWithRegisteringToAnotherModelGroup(modelId, newModelGroupId, user, updateModelInput, actionListener, context);
+                if (updateModelInput.getConnectorUpdateContent() != null) {
+                    Connector connector = mlModel.getConnector();
+                    connector.update(updateModelInput.getConnectorUpdateContent(), mlEngine::encrypt);
+                    connector.validateConnectorURL(trustedConnectorEndpointsRegex);
+                    updateModelInput.setConnector(connector);
+                    updateModelInput.setConnectorUpdateContent(null);
```

Comment:
Why set as null?

Reply:
We don't need this field after connector field is updated. To save some memory we set it back to null.